test/aleph/websocket_test.clj

```diff
@@ -88,14 +88,22 @@
       (is (= "YES" @(s/try-take! c 5e3))))))
 
 (deftest test-raw-echo-handler
-  (testing "websocket client: raw-stream?"
+  (testing "websocket client: raw-stream? with binary message"
     (with-handler echo-handler
       (let [c @(http/websocket-client "ws://localhost:8080" {:raw-stream? true})]
         (is @(s/put! c (.getBytes "raw client hello" "UTF-8")))
         (let [msg @(s/try-take! c 5e3)]
           (is (= "raw client hello"
                  (when msg (bs/to-string (netty/release-buf->array msg)))))))))
 
+  (testing "websocket client: raw-stream? with text message"
+    (with-handler echo-handler
+      (let [c @(http/websocket-client "ws://localhost:8080" {:raw-stream? true})]
+        (is @(s/put! c "text client hello"))
+        (let [msg @(s/try-take! c 5e3)]
+          (is (= "text client hello"
+                 (when msg (bs/to-string (netty/release-buf->array msg)))))))))
```

Comment:
I think we can do 

```clojure
(try (bs/to-string msg) (finally (netty/release msg)))
```

to avoid unnecessary conversion step.

Reply:
Strangely, this throws a reference count exception, indicating `msg` is has a zero ref count. Sure enough adding an acquire call in the client handler before putting it on the channel fixes that, but that's obviously not how it's supposed to work. 

The .content call `(.content ^TextWebSocketFrame msg)` should throw on zero ref count, so I can't see how it's getting changed?

Reply:
Aha, `bs/to-string` is calling release somewhere. Will update test.